pkg/kubectl/cmd/config/rename_context.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+// RenameContextOptions contains the options for running the rename-context cli command.
+type RenameContextOptions struct {
+	configAccess clientcmd.ConfigAccess
+	contextName  string
+	newName      string
+}
+
+const (
+	renameContextUse = "rename-context CONTEXT_NAME NEW_NAME"
+
+	renameContextShort = "Renames a context from the kubeconfig file."
+)
+
+var (
+	renameContextLong = templates.LongDesc(`
+		Renames a context from the kubeconfig file.
+
+		CONTEXT_NAME is the context name that you wish change.
+
+		NEW_NAME is the new name you wish to set.`)
```

Comment:
Add a note here in the long desc mentioning something like "in case the context being renamed is the one in use, the 'current-context' field will also be updated to point to the new name". Good job handling that btw.